st2common/bin/migrations/v3.5/st2-migrate-db-dict-field-values

```diff
@@ -0,0 +1,460 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Migration which which migrates data for existing objects in the database which utilize
+EscapedDictField or EscapedDynamicField and have been updated to use new JsonDictField.
+
+Migration step is idempotent and can be retried on failures / partial runs.
+
+Keep in mind that running this migration script is optional and it may take a long time of you have
+a lot of very large objects in the database (aka executions) - reading a lot of data from the
+database using the old field types is slow and CPU intensive.
+
+New field type is automatically used for all the new objects when upgrading to v3.5 so migration is
+optional because in most cases users are viewing recent / new executions and not old ones which may
+still utilize old field typo which is slow to read / write.
+
+Right now the script utilizes no concurrency and performs migration one object by one. That's done
+for simplicity reasons and also to avoid massive CPU usage spikes when running this script with
+large concurrency on large objects.
+
+Keep in mind that only "completed" objects are processes - this means Executions in "final" states
+(succeeded, failed, timeout, etc.).
+
+We determine if an object should be migrating using mongodb $type query (for execution objects we
+could also determine that based on the presence of result_size field).
+"""
+
+import sys
+import time
+import datetime
+import traceback
+
+from oslo_config import cfg
+
+from st2common import config
+from st2common.service_setup import db_setup
+from st2common.service_setup import db_teardown
+from st2common.util import isotime
+from st2common.models.db.execution import ActionExecutionDB
+from st2common.models.db.workflow import WorkflowExecutionDB
+from st2common.models.db.workflow import TaskExecutionDB
+from st2common.models.db.trigger import TriggerInstanceDB
+from st2common.persistence.execution import ActionExecution
+from st2common.persistence.liveaction import LiveAction
+from st2common.persistence.workflow import WorkflowExecution
+from st2common.persistence.workflow import TaskExecution
+from st2common.persistence.trigger import TriggerInstance
+from st2common.exceptions.db import StackStormDBObjectNotFoundError
+from st2common.constants.action import LIVEACTION_COMPLETED_STATES
+from st2common.constants.triggers import TRIGGER_INSTANCE_COMPLETED_STATES
+
+# We retrieve and process up to this amount of object IDs in a single MongoDB query. This is done
+# to avoid potentially retrieving many 10s of millions of object ids in a single query
+# TODO: Implement batching --since-ts, --model=action,trigger,workflow flag
+BATCH_SIZE = 1000
```

Comment:
Looks like batching is not yet implemented and I guess this variable definition is obsolete? I'll approve the PR to unblock you and all those looking forward for the fix on the apt cache issue as this is just a minor issue.

Reply:
Yep, this is not needed anymore - won't implement it since --start-dt and --end-dt was implemented which can be used to achieve similar end result.

Will remove it.